ksqldb-engine/src/main/java/io/confluent/ksql/engine/EngineExecutor.java

```diff
@@ -511,12 +507,28 @@ private boolean isSourceTableMaterializationEnabled() {
         .getBoolean(KsqlConfig.KSQL_SOURCE_TABLE_MATERIALIZATION_ENABLED);
   }
 
+  private SourceName getSourceName(final ConfiguredStatement<?> statement) {
+    if (statement.getStatement() instanceof CreateStream && ((CreateStream) statement.getStatement()).isNotExists()) {
+      return ((CreateStream) statement.getStatement()).getName();
+    } else if (statement.getStatement() instanceof CreateTable && ((CreateTable) statement.getStatement()).isNotExists()) {
+      return ((CreateTable) statement.getStatement()).getName();
+    } else if (statement.getStatement() instanceof CreateAsSelect && ((CreateAsSelect) statement.getStatement()).isNotExists()) {
+      return ((CreateAsSelect) statement.getStatement()).getName();
+    }
+    return null;
+  }
+
   // Known to be non-empty
   @SuppressWarnings("OptionalGetWithoutIsPresent")
   KsqlPlan plan(final ConfiguredStatement<?> statement) {
     try {
       throwOnNonExecutableStatement(statement);
 
+      final SourceName sourceName = getSourceName(statement);
+      if (sourceName != null && engineContext.getMetaStore().getSource(sourceName) != null) {
```

Comment:
you need to check that this is an `IF NOT EXISTS` statement and only ignore the missing source in that case